### ISSUE TITLE
mega-cmd-server crashes

### ISSUE THREAD

**Issue by srulikuk** [meganz/MEGAcmd#554]
Just installed on a new server, there are a few 100k of files to sync however it keeps on crashing at random times with nothing useful in the debug log

I ran it 10 times with debug to file, the smallest the file was when it crashed was 150MB, the largest 3.8GB, 
```
# date ; mega-cmd-server --debug-full >> mega_debug.log ; date
Sun  1 Aug 13:49:11 UTC 2021
Segmentation fault (core dumped)
Sun  1 Aug 14:33:42 UTC 2021
```
last 20 lines in debug file (replaced identifying names with xxx)
```
[API:debug: 14:33:42] Writing data asynchronously at 131072 to 144106
[API:debug: 14:33:42] Filesystem notification (deletion). Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification. Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification (deletion). Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification. Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification (deletion). Root: Mizrahi   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification. Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification (deletion). Root: Mizrahi   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Filesystem notification. Root: xxxx   Path: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
[API:debug: 14:33:42] Cached data at: 131072   Size: 19342
[API:debug: 14:33:42] Contiguous progress: 0 (150414)
[API:debug: 14:33:42] Caching transfer
[API:debug: 14:33:42] DB transaction BEGIN /root/.megaCmd/megaclient_statecache12_transfers_xxxx.db
[API:debug: 14:33:42] Cached data at: 0   Size: 59909
[API:debug: 14:33:42] Contiguous progress: 59909 (0)
[API:debug: 14:33:42] Caching transfer
[API:debug: 14:33:42] Download complete: SPZRgAaB 1
[API:debug: 14:33:42] Overwriting a local synced file. Moving the previous one to debris
[API:debug: 14:33:42] Renaming temporary file to target path
[API:info: 14:33:42] Transfer (DOWNLOAD) finished. File: xxxxxxx$_xxxxxx=xxxxx_xx$_x$_xxxxx=xxxxx$_B.jpg
```

how can I get to the bottom of this?

**Comment by polmr** [comment]
Hi @srulikuk , a stack trace would help us find where the crash is happening. Here are some instructions
(asuming debian based distro):

1 - install debug symbols
`sudo apt update && sudo apt install megacmd-dbg`

2 - run mega-cmd-server in a debugger (you might need to install gdb)
`gdb mega-cmd-server`
It should say something like:

```
done.
(gdb) 
```

3 - Then run the program (type 'r' + Enter)
Let it run until it crashes.

4 - once so, then type in the gdb console: 'thread apply all bt'
Press "Enter" until the output is complete, and

5 - then please, scroll back and send us the entire output of that command.

**Comment by srulikuk** [comment]
here it is

```
[Thread 0x7fffd37fe700 (LWP 131871) exited]

Thread 4 "mega-cmd-server" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4c62700 (LWP 131855)]
mega::LocalNode::getlocalpath (this=0xffffffffffffffff, path=..., sdisable=sdisable@entry=true, localseparator=localseparator@entry=0x0) at src/node.cpp:1620
1620	src/node.cpp: No such file or directory.
(gdb) thread apply all bt

Thread 21 (Thread 0x7fffd2ffd700 (LWP 131872)):
#0  0x00007ffff6f9a3bf in __GI___clock_nanosleep (clock_id=clock_id@entry=0, flags=flags@entry=0, req=req@entry=0x7fffd2ffca80, rem=rem@entry=0x0) at ../sysdeps/unix/sysv/linux/clock_nanosleep.c:78
#1  0x00007ffff6fa0047 in __GI___nanosleep (requested_time=requested_time@entry=0x7fffd2ffca80, remaining=remaining@entry=0x0) at nanosleep.c:27
#2  0x00007ffff6fd29bf in usleep (useconds=<optimized out>) at ../sysdeps/posix/usleep.c:32
#3  0x00005555556c5d9f in megacmd::retryConnections (pointer=<optimized out>) at src/megacmd.cpp:3765
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 19 (Thread 0x7fffd3fff700 (LWP 131870)):
#0  0x00007ffff6fd212b in __GI___select (nfds=68, readfds=readfds@entry=0x555557e92e30, writefds=writefds@entry=0x555557e92eb0, exceptfds=exceptfds@entry=0x555557e92f30, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557e92e20) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557e1fbc0) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557e1fbc0) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 18 (Thread 0x7fffe4ff9700 (LWP 131869)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557e98408) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557e983b8, cond=0x555557e983e0) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557e983e0, mutex=0x555557e983b8) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557e983e0) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557e983b0) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 17 (Thread 0x7fffe57fa700 (LWP 131868)):
#0  0x00007ffff6fd212b in __GI___select (nfds=69, readfds=readfds@entry=0x555557e93150, writefds=writefds@entry=0x555557e931d0, exceptfds=exceptfds@entry=0x555557e93250, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557e93140) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557e93130) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 16 (Thread 0x7fffe5ffb700 (LWP 131867)):
#0  0x00007ffff6fd212b in __GI___select (nfds=57, readfds=readfds@entry=0x555557e16ca0, writefds=writefds@entry=0x555557e16d20, exceptfds=exceptfds@entry=0x555557e16da0, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557e16c90) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557da3a30) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557da3a30) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 15 (Thread 0x7fffe67fc700 (LWP 131866)):
--Type <RET> for more, q to quit, c to continue without paging--
```

**Comment by bashfulrobot** [comment]
I too am running into this. I am including my output as well. If this needs to be split into another issue, please let me know.

## Versions

```
❯ cat -p /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=21.04
DISTRIB_CODENAME=hirsute
DISTRIB_DESCRIPTION="Ubuntu 21.04"
```

```
MEGAcmd version: 1.4.0.1: code 1040001
```


## Debugging Output

```
(gdb) thread apply all bt

Thread 22 (Thread 0x7ffff43af980 (LWP 828511) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x7ffff43aeed0, clockid=0, expected=0, futex_word=0x7ffff6f042e8 <__aio_new_request_notification+40>) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x7ffff6f042e8 <__aio_new_request_notification+40>, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x7ffff43aeed0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff730899e in __pthread_cond_wait_common (abstime=0x7ffff43aeed0, clockid=0, mutex=0x7ffff6f04220 <__aio_requests_mutex>, cond=0x7ffff6f042c0 <__aio_new_request_notification>) at pthread_cond_wait.c:504
#3  __pthread_cond_timedwait (cond=cond@entry=0x7ffff6f042c0 <__aio_new_request_notification>, mutex=mutex@entry=0x7ffff6f04220 <__aio_requests_mutex>, abstime=abstime@entry=0x7ffff43aeed0) at pthread_cond_wait.c:646
#4  0x00007ffff6efdfad in handle_fildes_io (arg=<optimized out>) at ../sysdeps/pthread/aio_misc.c:629
#5  0x00007ffff7302450 in start_thread (arg=0x7ffff43af980) at pthread_create.c:473
#6  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 21 (Thread 0x7fffda7fc640 (LWP 828400) "mega-cmd-server"):
#0  0x00007ffff6a5b5df in __GI___clock_nanosleep (clock_id=clock_id@entry=0, flags=flags@entry=0, req=req@entry=0x7fffda7fb9e0, rem=rem@entry=0x0) at ../sysdeps/unix/sysv/linux/clock_nanosleep.c:73
#1  0x00007ffff6a60ac7 in __GI___nanosleep (req=req@entry=0x7fffda7fb9e0, rem=rem@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:25
#2  0x00007ffff6a923c9 in usleep (useconds=useconds@entry=300000) at ../sysdeps/posix/usleep.c:32
#3  0x000055555568fc4f in megacmd::sleepMilliSeconds (milliseconds=milliseconds@entry=300) at src/megacmdcommonutils.cpp:955
#4  0x000055555566a8af in megacmd::retryConnections (pointer=<optimized out>) at src/megacmd.cpp:3765
#5  0x00007ffff7302450 in start_thread (arg=0x7fffda7fc640) at pthread_create.c:473
#6  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 19 (Thread 0x7fffdb7fe640 (LWP 828398) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=68, readfds=readfds@entry=0x555557d2b630, writefds=writefds@entry=0x555557d2b6b0, exceptfds=exceptfds@entry=0x555557d2b730, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=0x555557d2b620) at src/posix/waiter.cpp:113
#2  0x00005555557f61ff in mega::MegaApiImpl::loop (this=0x555557cb84c0) at src/megaapi_impl.cpp:6279
#3  0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x555557cb84c0) at src/megaapi_impl.cpp:5038
#4  0x00007ffff7302450 in start_thread (arg=0x7fffdb7fe640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 18 (Thread 0x7fffdbfff640 (LWP 828397) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557d30bf0) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557d30bf0, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557d30ba0, cond=0x555557d30bc8) at pthread_cond_wait.c:504
#3  __pthread_cond_wait (cond=0x555557d30bc8, mutex=0x555557d30ba0) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557d30bc8) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557d30b98) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7fffdbfff640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 17 (Thread 0x7fffe4ff9640 (LWP 828396) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=69, readfds=readfds@entry=0x555557d2b950, writefds=writefds@entry=0x555557d2b9d0, exceptfds=exceptfds@entry=0x555557d2ba50, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557d2b940) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557d2b930) at src/gfx.cpp:106
--Type <RET> for more, q to quit, c to continue without paging--
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7fffe4ff9640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 16 (Thread 0x7fffe57fa640 (LWP 828395) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=57, readfds=readfds@entry=0x555557caf5c0, writefds=writefds@entry=0x555557caf640, exceptfds=exceptfds@entry=0x555557caf6c0, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=0x555557caf5b0) at src/posix/waiter.cpp:113
#2  0x00005555557f61ff in mega::MegaApiImpl::loop (this=0x555557c3c450) at src/megaapi_impl.cpp:6279
#3  0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x555557c3c450) at src/megaapi_impl.cpp:5038
#4  0x00007ffff7302450 in start_thread (arg=0x7fffe57fa640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 15 (Thread 0x7fffe5ffb640 (LWP 828394) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557cb4b80) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557cb4b80, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557cb4b30, cond=0x555557cb4b58) at pthread_cond_wait.c:504
#3  __pthread_cond_wait (cond=0x555557cb4b58, mutex=0x555557cb4b30) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557cb4b58) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557cb4b28) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7fffe5ffb640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 14 (Thread 0x7fffe67fc640 (LWP 828393) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=58, readfds=readfds@entry=0x555557caf8e0, writefds=writefds@entry=0x555557caf960, exceptfds=exceptfds@entry=0x555557caf9e0, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557caf8d0) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557caf8c0) at src/gfx.cpp:106
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7fffe67fc640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 13 (Thread 0x7fffe6ffd640 (LWP 828392) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=46, readfds=readfds@entry=0x555557c33550, writefds=writefds@entry=0x555557c335d0, exceptfds=exceptfds@entry=0x555557c33650, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=0x555557c33540) at src/posix/waiter.cpp:113
#2  0x00005555557f61ff in mega::MegaApiImpl::loop (this=0x555557bc03e0) at src/megaapi_impl.cpp:6279
#3  0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x555557bc03e0) at src/megaapi_impl.cpp:5038
#4  0x00007ffff7302450 in start_thread (arg=0x7fffe6ffd640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 12 (Thread 0x7fffe77fe640 (LWP 828391) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557c38b10) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557c38b10, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557c38ac0, cond=0x555557c38ae8) at pthread_cond_wait.c:504
--Type <RET> for more, q to quit, c to continue without paging--
#3  __pthread_cond_wait (cond=0x555557c38ae8, mutex=0x555557c38ac0) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557c38ae8) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557c38ab8) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7fffe77fe640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 11 (Thread 0x7fffe7fff640 (LWP 828390) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=47, readfds=readfds@entry=0x555557c33870, writefds=writefds@entry=0x555557c338f0, exceptfds=exceptfds@entry=0x555557c33970, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557c33860) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557c33850) at src/gfx.cpp:106
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7fffe7fff640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 10 (Thread 0x7fffecff9640 (LWP 828389) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=35, readfds=readfds@entry=0x555557bb74e0, writefds=writefds@entry=0x555557bb7560, exceptfds=exceptfds@entry=0x555557bb75e0, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=0x555557bb74d0) at src/posix/waiter.cpp:113
#2  0x00005555557f61ff in mega::MegaApiImpl::loop (this=0x555557b44370) at src/megaapi_impl.cpp:6279
#3  0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x555557b44370) at src/megaapi_impl.cpp:5038
#4  0x00007ffff7302450 in start_thread (arg=0x7fffecff9640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 9 (Thread 0x7fffed7fa640 (LWP 828388) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557bbcaa0) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557bbcaa0, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557bbca50, cond=0x555557bbca78) at pthread_cond_wait.c:504
#3  __pthread_cond_wait (cond=0x555557bbca78, mutex=0x555557bbca50) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557bbca78) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557bbca48) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7fffed7fa640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 8 (Thread 0x7fffedffb640 (LWP 828387) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=36, readfds=readfds@entry=0x555557bb7800, writefds=writefds@entry=0x555557bb7880, exceptfds=exceptfds@entry=0x555557bb7900, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557bb77f0) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557bb77e0) at src/gfx.cpp:106
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7fffedffb640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 7 (Thread 0x7fffee7fc640 (LWP 828386) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=24, readfds=readfds@entry=0x555557b3b590, writefds=writefds@entry=0x555557b3b610, exceptfds=exceptfds@entry=0x555557b3b690, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/s--Type <RET> for more, q to quit, c to continue without paging--
elect.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=0x555557b3b580) at src/posix/waiter.cpp:113
#2  0x00005555557f61ff in mega::MegaApiImpl::loop (this=0x555557ac8540) at src/megaapi_impl.cpp:6279
#3  0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x555557ac8540) at src/megaapi_impl.cpp:5038
#4  0x00007ffff7302450 in start_thread (arg=0x7fffee7fc640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 6 (Thread 0x7fffeeffd640 (LWP 828385) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557b40a90) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557b40a90, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557b40a40, cond=0x555557b40a68) at pthread_cond_wait.c:504
#3  __pthread_cond_wait (cond=0x555557b40a68, mutex=0x555557b40a40) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557b40a68) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557b40a38) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7fffeeffd640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 5 (Thread 0x7fffef7fe640 (LWP 828384) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=25, readfds=readfds@entry=0x555557b3b7f0, writefds=writefds@entry=0x555557b3b870, exceptfds=exceptfds@entry=0x555557b3b8f0, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557b3b7e0) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557b3b7d0) at src/gfx.cpp:106
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7fffef7fe640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
^M^M
Thread 4 (Thread 0x7fffeffff640 (LWP 828383) "mega-cmd-server"):
#0  mega::LocalNode::getlocalpath (this=0xffffffffffffffff, path=..., sdisable=sdisable@entry=true, localseparator=localseparator@entry=0x0) at src/node.cpp:1620
#1  0x00005555557a9e41 in mega::MegaNodePrivate::MegaNodePrivate (this=0x7fffbba87460, node=0x7fffbba6fa80) at src/megaapi_impl.cpp:463
#2  0x00005555557aac4b in mega::MegaNodePrivate::fromNode (node=0x7fffbba6fa80) at src/megaapi_impl.cpp:2112
#3  0x00005555557ad2de in mega::MegaApiImpl::getNodeByHandle (this=0x5555579ed380, handle=91329265938160) at src/megaapi_impl.cpp:17779
#4  0x000055555580307d in mega::MegaApi::getNodeByHandle (this=this@entry=0x55555799b160, h=<optimized out>) at src/megaapi.cpp:4005
#5  0x00005555556af665 in megacmd::MegaCmdGlobalTransferListener::onTransferFinish (this=0x555557d34530, api=0x55555799b160, transfer=0x7fffbbabf990, error=<optimized out>) at src/listeners.cpp:1041
#6  0x0000555555781c3b in mega::MegaApiImpl::fireOnTransferFinish (this=0x5555579ed380, transfer=0x7fffbbabf990, e=std::unique_ptr<mega::MegaErrorPrivate> = {...}, committer=...) at src/megaapi_impl.cpp:16043
#7  0x000055555578777f in mega::MegaApiImpl::processTransferComplete (this=0x5555579ed380, tr=0x7fffc80087e0, transfer=0x7fffbbabf990) at src/megaapi_impl.cpp:16505
#8  0x00005555558afd04 in mega::Transfer::complete (this=0x7fffc80087e0, committer=...) at src/transfer.cpp:918
#9  0x00005555558b4fb7 in mega::TransferSlot::checkTransferFinished (client=<optimized out>, committer=..., this=<optimized out>) at src/transferslot.cpp:367
#10 mega::TransferSlot::checkTransferFinished (this=0x7fffcc0c6160, committer=..., client=0x555557a77be0) at src/transferslot.cpp:344
#11 0x00005555558b709a in mega::TransferSlot::doio (this=0x7fffcc0c6160, client=client@entry=0x555557a77be0, committer=...) at src/transferslot.cpp:802
#12 0x0000555555867a98 in mega::MegaClient::exec (this=<optimized out>) at src/megaclient.cpp:2369
#13 0x00005555557f628e in mega::MegaApiImpl::loop (this=0x5555579ed380) at src/megaapi_impl.cpp:6301
#14 0x00005555557f6361 in mega::MegaApiImpl::threadEntryPoint (param=0x5555579ed380) at src/megaapi_impl.cpp:5038--Type <RET> for more, q to quit, c to continue without paging--

#15 0x00007ffff7302450 in start_thread (arg=0x7fffeffff640) at pthread_create.c:473
#16 0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 3 (Thread 0x7ffff4c29640 (LWP 828382) "mega-cmd-server"):
#0  __futex_abstimed_wait_common64 (cancel=true, private=0, abstime=0x0, clockid=0, expected=0, futex_word=0x555557a7baf4) at ../sysdeps/nptl/futex-internal.c:74
#1  __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555557a7baf4, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ../sysdeps/nptl/futex-internal.c:123
#2  0x00007ffff7308540 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557a7baa0, cond=0x555557a7bac8) at pthread_cond_wait.c:504
#3  __pthread_cond_wait (cond=0x555557a7bac8, mutex=0x555557a7baa0) at pthread_cond_wait.c:628
#4  0x00007ffff6dac280 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x000055555574540b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557a7bac8) at /usr/include/c++/10/condition_variable:111
#6  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557a7ba98) at src/utils.cpp:2433
#7  0x00007ffff6db2694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff7302450 in start_thread (arg=0x7ffff4c29640) at pthread_create.c:473
#9  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 2 (Thread 0x7ffff542a640 (LWP 828381) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=14, readfds=readfds@entry=0x555557a76d60, writefds=writefds@entry=0x555557a76de0, exceptfds=exceptfds@entry=0x555557a76e60, timeout=<optimized out>) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x00005555558263c3 in mega::PosixWaiter::wait (this=this@entry=0x555557a76d50) at src/posix/waiter.cpp:113
#2  0x000055555589d0d4 in mega::GfxProc::loop (this=0x555557a76d40) at src/gfx.cpp:106
#3  0x000055555589d64d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff7302450 in start_thread (arg=0x7ffff542a640) at pthread_create.c:473
#5  0x00007ffff6a9ad53 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 1 (Thread 0x7ffff59a1980 (LWP 828352) "mega-cmd-server"):
#0  0x00007ffff6a91b31 in __GI___select (nfds=nfds@entry=1024, readfds=readfds@entry=0x555557d347c0, writefds=writefds@entry=0x0, exceptfds=exceptfds@entry=0x0, timeout=timeout@entry=0x0) at ../sysdeps/unix/sysv/linux/select.c:49
#1  0x0000555555707c02 in megacmd::ComunicationsManagerFileSockets::waitForPetition (this=0x555557d34720) at src/comunicationsmanagerfilesockets.cpp:214
#2  0x0000555555675891 in megacmd::megacmd () at src/megacmd.cpp:3918
#3  0x000055555565c750 in main (argc=<optimized out>, argv=<optimized out>) at src/megacmd.cpp:5014
```

**Comment by polmr** [comment]
Thanks guys!
Note: @srulikuk output didn't have the entire output (with the offending thread trace) and @bashfulrobot's didn't have which thread segfaulted (anyway it's thread 4 too ;-) ).
Here comes the interesting part:
```Thread 4 (Thread 0x7fffeffff640 (LWP 828383) "mega-cmd-server"):
#0  mega::LocalNode::getlocalpath (this=0xffffffffffffffff, path=..., sdisable=sdisable@entry=true, localseparator=localseparator@entry=0x0) at src/node.cpp:1620
#1  0x00005555557a9e41 in mega::MegaNodePrivate::MegaNodePrivate (this=0x7fffbba87460, node=0x7fffbba6fa80) at src/megaapi_impl.cpp:463
```
`this=0xffffffffffffffff` -> this special value should never be accessed.
This is very likely solved in newer SDK versions than the one used in current MEGAcmd version. We will soon release a new version that will probably fix that.  Please, stay tuned and apologies for the inconveniences.

**Comment by srulikuk** [comment]
@polmr sorry about that, here is my full output, i al so seem to have `this=0xffffffffffffffff` in my output. in how long do you expect to have a fix and do you have a work around? (I am on ubuntu 20.04.2 LTS)

```
Thread 4 "mega-cmd-server" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4c62700 (LWP 169933)]
mega::LocalNode::getlocalpath (this=0xffffffffffffffff, path=..., sdisable=sdisable@entry=true, localseparator=localseparator@entry=0x0) at src/node.cpp:1620
1620	src/node.cpp: No such file or directory.
(gdb) thread apply all bt

Thread 21 (Thread 0x7fffd2ffd700 (LWP 169950)):
#0  0x00007ffff6f9a3bf in __GI___clock_nanosleep (clock_id=clock_id@entry=0, flags=flags@entry=0, req=req@entry=0x7fffd2ffca80, rem=rem@entry=0x0) at ../sysdeps/unix/sysv/linux/clock_nanosleep.c:78
#1  0x00007ffff6fa0047 in __GI___nanosleep (requested_time=requested_time@entry=0x7fffd2ffca80, remaining=remaining@entry=0x0) at nanosleep.c:27
#2  0x00007ffff6fd29bf in usleep (useconds=<optimized out>) at ../sysdeps/posix/usleep.c:32
#3  0x00005555556c5d9f in megacmd::retryConnections (pointer=<optimized out>) at src/megacmd.cpp:3765
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 20 (Thread 0x7fffd37fe700 (LWP 169949)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x7fffcc00bb60) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x7fffcc00bb10, cond=0x7fffcc00bb38) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=cond@entry=0x7fffcc00bb38, mutex=mutex@entry=0x7fffcc00bb10) at pthread_cond_wait.c:638
#3  0x0000555555876cdb in mega::PosixSemaphore::wait (this=0x7fffcc00bb00) at src/thread/posixthread.cpp:89
#4  0x00005555557273be in megacmd::MegaCmdExecuter::checkAndInformPSA (this=0x555557b59850, inf=0x0, enforce=<optimized out>) at src/megacmdexecuter.cpp:949
#5  0x000055555573987a in megacmd::MegaCmdExecuter::fetchNodes (this=0x555557b59850, api=0x555557b07b00, clientID=<optimized out>) at src/megacmdexecuter.cpp:2514
#6  0x000055555573dd8c in megacmd::MegaCmdExecuter::actUponLogin (this=0x555557b59850, srl=0x7fffcc001600, timeout=<optimized out>) at src/megacmdexecuter.cpp:2410
#7  0x000055555574df7d in megacmd::MegaCmdExecuter::executecommand (this=0x555557b59850, words=std::vector of length 2, capacity 2 = {...}, clflags=0x7fffd37fd590, cloptions=0x7fffd37fd560) at src/megacmdexecuter.cpp:8065
#8  0x00005555556df874 in megacmd::executecommand (ptr=<optimized out>) at src/megacmd.cpp:3108
#9  0x00005555556e1121 in megacmd::process_line (l=0x555557e9cf90 "login AV8CrR4dBAICPlvWok6Z4Ri_jq1Hq8m5t_sIv9rLU5BhTGNVclNhZDNHVne_sNFkD34cFi9m3I5F6BXV") at src/megacmd.cpp:3529
#10 megacmd::doProcessLine (pointer=0x555557e99a20) at src/megacmd.cpp:3565
#11 0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#12 0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 19 (Thread 0x7fffd3fff700 (LWP 169948)):
#0  0x00007ffff6fd212b in __GI___select (nfds=68, readfds=readfds@entry=0x555557e92e30, writefds=writefds@entry=0x555557e92eb0, exceptfds=exceptfds@entry=0x555557e92f30, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557e92e20) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557e1fbc0) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557e1fbc0) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 18 (Thread 0x7fffe4ff9700 (LWP 169947)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557e98408) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557e983b8, cond=0x555557e983e0) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557e983e0, mutex=0x555557e983b8) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557e983e0) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557e983b0) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 17 (Thread 0x7fffe57fa700 (LWP 169946)):
#0  0x00007ffff6fd212b in __GI___select (nfds=69, readfds=readfds@entry=0x555557e93150, writefds=writefds@entry=0x555557e931d0, exceptfds=exceptfds@entry=0x555557e93250, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
--Type <RET> for more, q to quit, c to continue without paging--c
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557e93140) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557e93130) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 16 (Thread 0x7fffe5ffb700 (LWP 169945)):
#0  0x00007ffff6fd212b in __GI___select (nfds=57, readfds=readfds@entry=0x555557e16ca0, writefds=writefds@entry=0x555557e16d20, exceptfds=exceptfds@entry=0x555557e16da0, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557e16c90) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557da3a30) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557da3a30) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 15 (Thread 0x7fffe67fc700 (LWP 169944)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557e1c278) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557e1c228, cond=0x555557e1c250) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557e1c250, mutex=0x555557e1c228) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557e1c250) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557e1c220) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 14 (Thread 0x7fffe6ffd700 (LWP 169943)):
#0  0x00007ffff6fd212b in __GI___select (nfds=58, readfds=readfds@entry=0x555557e16fc0, writefds=writefds@entry=0x555557e17040, exceptfds=exceptfds@entry=0x555557e170c0, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557e16fb0) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557e16fa0) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 13 (Thread 0x7fffe77fe700 (LWP 169942)):
#0  0x00007ffff6fd212b in __GI___select (nfds=46, readfds=readfds@entry=0x555557d9ab10, writefds=writefds@entry=0x555557d9ab90, exceptfds=exceptfds@entry=0x555557d9ac10, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557d9ab00) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557d278a0) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557d278a0) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 12 (Thread 0x7fffe7fff700 (LWP 169941)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557da00e8) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557da0098, cond=0x555557da00c0) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557da00c0, mutex=0x555557da0098) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557da00c0) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557da0090) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 11 (Thread 0x7fffecff9700 (LWP 169940)):
#0  0x00007ffff6fd212b in __GI___select (nfds=47, readfds=readfds@entry=0x555557d9ae30, writefds=writefds@entry=0x555557d9aeb0, exceptfds=exceptfds@entry=0x555557d9af30, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557d9ae20) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557d9ae10) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 10 (Thread 0x7fffed7fa700 (LWP 169939)):
#0  0x00007ffff6fd212b in __GI___select (nfds=35, readfds=readfds@entry=0x555557d1e980, writefds=writefds@entry=0x555557d1ea00, exceptfds=exceptfds@entry=0x555557d1ea80, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557d1e970) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557cab710) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557cab710) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 9 (Thread 0x7fffedffb700 (LWP 169938)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557d23f58) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557d23f08, cond=0x555557d23f30) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557d23f30, mutex=0x555557d23f08) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557d23f30) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557d23f00) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 8 (Thread 0x7fffee7fc700 (LWP 169937)):
#0  0x00007ffff6fd212b in __GI___select (nfds=36, readfds=readfds@entry=0x555557d1eca0, writefds=writefds@entry=0x555557d1ed20, exceptfds=exceptfds@entry=0x555557d1eda0, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557d1ec90) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557d1ec80) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 7 (Thread 0x7fffeeffd700 (LWP 169936)):
#0  0x00007ffff6fd212b in __GI___select (nfds=24, readfds=readfds@entry=0x555557ca2910, writefds=writefds@entry=0x555557ca2990, exceptfds=exceptfds@entry=0x555557ca2a10, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=0x555557ca2900) at src/posix/waiter.cpp:113
#2  0x0000555555846d2f in mega::MegaApiImpl::loop (this=0x555557c2f790) at src/megaapi_impl.cpp:6279
#3  0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557c2f790) at src/megaapi_impl.cpp:5038
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 6 (Thread 0x7fffef7fe700 (LWP 169935)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557ca7e28) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557ca7dd8, cond=0x555557ca7e00) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557ca7e00, mutex=0x555557ca7dd8) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557ca7e00) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557ca7dd0) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 5 (Thread 0x7fffeffff700 (LWP 169934)):
#0  0x00007ffff6fd212b in __GI___select (nfds=25, readfds=readfds@entry=0x555557ca2b70, writefds=writefds@entry=0x555557ca2bf0, exceptfds=exceptfds@entry=0x555557ca2c70, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557ca2b60) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557ca2b50) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 4 (Thread 0x7ffff4c62700 (LWP 169933)):
#0  mega::LocalNode::getlocalpath (this=0xffffffffffffffff, path=..., sdisable=sdisable@entry=true, localseparator=localseparator@entry=0x0) at src/node.cpp:1620
#1  0x0000555555806fab in mega::MegaNodePrivate::MegaNodePrivate (this=0x7fffa3781320, node=0x7fffac873d60) at src/megaapi_impl.cpp:463
#2  0x0000555555807ccb in mega::MegaNodePrivate::fromNode (node=0x7fffac873d60) at src/megaapi_impl.cpp:2112
#3  0x0000555555808267 in mega::MegaNodeListPrivate::MegaNodeListPrivate (this=0x7fffaf902b50, newlist=0x7fffa3773da0, size=<optimized out>) at src/megaapi_impl.cpp:4186
#4  0x0000555555829819 in mega::MegaApiImpl::getChildren (this=<optimized out>, p=<optimized out>, order=order@entry=1) at /usr/include/c++/9/bits/stl_vector.h:915
#5  0x0000555555855e7d in mega::MegaApi::getChildren (this=this@entry=0x555557b07b00, p=<optimized out>, order=order@entry=1) at src/megaapi.cpp:3950
#6  0x00005555556e8f57 in megacmd::getNumFolderFiles (n=<optimized out>, api=0x555557b07b00, nfiles=0x7ffff4c61208, nfolders=0x7ffff4c61200) at src/megacmdutils.cpp:41
#7  0x00005555556e8fb2 in megacmd::getNumFolderFiles (n=<optimized out>, api=0x555557b07b00, nfiles=0x7ffff4c61208, nfolders=0x7ffff4c61200) at src/megacmdutils.cpp:51
#8  0x00005555556e8fb2 in megacmd::getNumFolderFiles (n=<optimized out>, api=0x555557b07b00, nfiles=0x7ffff4c61208, nfolders=0x7ffff4c61200) at src/megacmdutils.cpp:51
#9  0x00005555556e8fb2 in megacmd::getNumFolderFiles (n=<optimized out>, api=0x555557b07b00, nfiles=0x7ffff4c61208, nfolders=0x7ffff4c61200) at src/megacmdutils.cpp:51
#10 0x00005555556e8fb2 in megacmd::getNumFolderFiles (n=<optimized out>, api=0x555557b07b00, nfiles=0x7ffff4c61208, nfolders=0x7ffff4c61200) at src/megacmdutils.cpp:51
#11 0x00005555556e8fb2 in megacmd::getNumFolderFiles (n=n@entry=0x7fffa376a400, api=api@entry=0x555557b07b00, nfiles=nfiles@entry=0x7ffff4c61208, nfolders=nfolders@entry=0x7ffff4c61200) at src/megacmdutils.cpp:51
#12 0x0000555555707ab3 in megacmd::MegaCmdGlobalListener::onNodesUpdate (this=<optimized out>, api=0x555557b07b00, nodes=<optimized out>) at src/listeners.cpp:115
#13 0x00005555557cabbf in mega::MegaApiImpl::fireOnNodesUpdate (this=this@entry=0x555557b59ed0, nodes=nodes@entry=0x0) at /usr/include/c++/9/bits/stl_tree.h:371
#14 0x000055555580868a in mega::MegaApiImpl::nodes_updated (this=0x555557b59ed0, n=0x0, count=420529) at src/megaapi_impl.cpp:14549
#15 0x00005555558ab436 in mega::MegaClient::procsc (this=<optimized out>) at ./include/mega/db.h:140
#16 0x00005555558b1627 in mega::MegaClient::exec (this=<optimized out>) at src/megaclient.cpp:2214
#17 0x0000555555846dbe in mega::MegaApiImpl::loop (this=0x555557b59ed0) at src/megaapi_impl.cpp:6301
#18 0x0000555555846e91 in mega::MegaApiImpl::threadEntryPoint (param=0x555557b59ed0) at src/megaapi_impl.cpp:5038
#19 0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#20 0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 3 (Thread 0x7ffff5463700 (LWP 169932)):
#0  futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x555557be8138) at ../sysdeps/nptl/futex-internal.h:183
#1  __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x555557be80e8, cond=0x555557be8110) at pthread_cond_wait.c:508
#2  __pthread_cond_wait (cond=0x555557be8110, mutex=0x555557be80e8) at pthread_cond_wait.c:638
#3  0x00007ffff72e8e30 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00005555557a515b in std::condition_variable::wait<mega::MegaClientAsyncQueue::asyncThreadLoop()::<lambda()> > (__p=..., __lock=..., this=0x555557be8110) at /usr/include/c++/9/bits/stl_deque.h:1384
#5  mega::MegaClientAsyncQueue::asyncThreadLoop (this=0x555557be80e0) at src/utils.cpp:2433
#6  0x00007ffff72eede4 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#8  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 2 (Thread 0x7ffff5c64700 (LWP 169931)):
#0  0x00007ffff6fd212b in __GI___select (nfds=14, readfds=readfds@entry=0x555557be3390, writefds=writefds@entry=0x555557be3410, exceptfds=exceptfds@entry=0x555557be3490, timeout=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x0000555555876803 in mega::PosixWaiter::wait (this=this@entry=0x555557be3380) at src/posix/waiter.cpp:113
#2  0x00005555558e4f7d in mega::GfxProc::loop (this=0x555557be3370) at src/gfx.cpp:106
#3  0x00005555558e568d in mega::GfxProc::threadEntryPoint (param=<optimized out>) at src/gfx.cpp:96
#4  0x00007ffff77fd609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#5  0x00007ffff6fdc293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 1 (Thread 0x7ffff5f4e600 (LWP 169927)):
#0  0x00007ffff6fd212b in __GI___select (nfds=nfds@entry=1024, readfds=readfds@entry=0x555557e9bfe0, writefds=writefds@entry=0x0, exceptfds=exceptfds@entry=0x0, timeout=timeout@entry=0x0) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x000055555576a5c4 in megacmd::ComunicationsManagerFileSockets::waitForPetition (this=0x555557e9bf40) at src/comunicationsmanagerfilesockets.cpp:214
#2  0x00005555556d6d71 in megacmd::megacmd () at src/megacmd.cpp:3918
#3  0x00005555556ba11c in main (argc=<optimized out>, argv=<optimized out>) at src/megacmd.cpp:5014
(gdb) 
```

**Comment by srulikuk** [comment]
@polmr do you have an ETA?

**Comment by bashfulrobot** [comment]
Hi there, I'm just following up on this. I was considering a subscription, but this tool is critical for me and my use cases.

Thank you very much for your time.

**Comment by srulikuk** [comment]
@polmr any update?

**Comment by bashfulrobot** [comment]
> Hi!
> We don't have a release date yet (it will be soon), but we can share with you the 1.5.0 RC1 packages which may fix your problems:
> https://mega.nz/folder/YsgSlCZL#iQYQo4Fkw5DimKxP9KIwDA
> 
> Install it on top of the one you currently have if any:
> 
> ```
> sudo apt install /path/to/downloaded/megacmd_1.5.0-1.1_amd64.deb
> ```
> 
> If you experience any crash, follow the instructions Pablo gave you above, but installing the `megacmd-dbg_1.5.0-1.1_amd64.deb` package in this link. Again, you can install it on top of the one you have.
> 
> If you install them, once the official 1.5.0 package is ready in our repositories, the package will be updated as any other package in your distro.
> No major problems are expected with this packages, but if you find any, please report! =)

I am just starting my sync tests back up, but so far, nothing seems to be failing thus far. :+1:

**Comment by srulikuk** [comment]
@aabellagm I tried to install megasync-xUbuntu_18.04_amd64.deb however i get dependency issues
```
The following packages have unmet dependencies:
 megasync : Depends: libraw16 (>= 0.16.0) but it is not installable
```
I installed libraw19 but it did not resolve the issue.

**Comment by aabellagm** [comment]
Hi @srulikuk I think you are using Ubuntu 20.04 and you are trying to install the package for Ubuntu 18.04.

In the link above, select xUbuntu_20.04 folder, then amd64 and download and install the package called `megacmd_1.5.0-1.1_amd64.deb`.

**Comment by srulikuk** [comment]
@aabellagm ooops, my bad. thanks

**Comment by srulikuk** [comment]
@polmr ^ ?

**Comment by polmr** [comment]
Hi @srulikuk , 
apologies for the late response.
It seems as if MEGAcmd was not receiving file system notifications for those files. The sync engine registers folders for inotify events, which should happen whenever a file changes/is created. 
Would you care to tell us what is your setup?
What do you mean exactly with `files which come in via rsync`?
Are you synching regular folders? or a network share?
A log will confirm if MEGAcmd is receiving these file system notifications.

**Comment by srulikuk** [comment]
@polmr 
It makes sense that this is an issue with inotify, however I only have this issue in 1 dir/sync all the sub dir's here are named by date_time and contain a single tar.gz file 100-300mb in size, the other syncs do not have this issue (the files there are small 1kb-10mb)

The files are on a backup server and each hour files are rsync'ed to this backup server.

My setup is a Ubuntu 20.04.3 LTS server, the sync dir's are NFS shares however this should not really have any effect as this server is the NFS host so the syncs are local dir's.

**Comment by srulikuk** [comment]
@polmr ^?

**Comment by mrtnggnn** [comment]
Fast-forward to 2025, similar problem.

OS: Void Linux
pkgname: MEGAcmd
pkgver: MEGAcmd-1.4.0_2

Stack trace shows it happens in mega::Node::serialize():
```
Thread 4 "mega-cmd-server" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff39fd6c0 (LWP 10249)]
__memcpy_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:342
warning: 342	../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S: No such file or directory
(gdb) thread apply all bt
...
Thread 4 (Thread 0x7ffff39fd6c0 (LWP 10249) "mega-cmd-server"):
#0  __memcpy_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:342
#1  0x00007ffff77882c3 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned long) () from /usr/lib/libstdc++.so.6
#2  0x00007ffff7bdedfb in mega::Node::serialize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) () from /usr/lib/libmega.so.30703
#3  0x00007ffff7bc32b9 in mega::DbTable::put(unsigned int, mega::Cacheable*, mega::SymmCipher*) () from /usr/lib/libmega.so.30703
#4  0x00007ffff7b3b273 in mega::MegaClient::updatesc() () from /usr/lib/libmega.so.30703
#5  0x00007ffff7b60d4b in mega::MegaClient::notifypurge() () from /usr/lib/libmega.so.30703
#6  0x00007ffff7b62d12 in mega::MegaClient::procsc() () from /usr/lib/libmega.so.30703
#7  0x00007ffff7b69de1 in mega::MegaClient::exec() () from /usr/lib/libmega.so.30703
#8  0x00007ffff7cd25f9 in mega::MegaApiImpl::loop() () from /usr/lib/libmega.so.30703
#9  0x00007ffff7cd26ad in mega::MegaApiImpl::threadEntryPoint(void*) () from /usr/lib/libmega.so.30703
#10 0x00007ffff74a2962 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:447
#11 0x00007ffff751db08 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:78
```
Any help available on this issue?
Let me know if more info is needed.

**Comment by mrtnggnn** [comment]
> pkgver: MEGAcmd-1.4.0_2

I see the Void package is somewhat outdated... Will build from src
